eegnb/cli/utils.py

```diff
@@ -1,111 +1,32 @@
-    elif experiment == "auditory-SSAEP onefreq":
-        ssaep_onefreq.present(duration=record_duration, eeg=eeg_device, save_fn=save_fn)
-    elif experiment == "auditory-oddball orig":
-        aob.present(duration=record_duration, eeg=eeg_device, save_fn=save_fn)
-    elif experiment == "auditory-oddball diaconescu":
-        F = h5py.File(mcond_file, "r")  # ['museEEG']
-        highPE = np.squeeze(F["museEEG"]["design"]["highPE"][:]).astype(int)
-        lowPE = np.squeeze(F["museEEG"]["design"]["lowPE"][:]).astype(int)
-        inputs = np.squeeze(F["museEEG"]["design"]["inputs"][:]).astype(int)
-
-        # based on inputs, creating oddball paradigms markers depending on "switch"
-        tonenums = maketonesnums(1800)
-        oddball3 = makeoddball(inputs, 3)
-        oddball4 = makeoddball(inputs, 4)
-        oddball5 = makeoddball(inputs, 5)
-        oddball6 = makeoddball(inputs, 6)
-
-        # modifying 0s in PE definitions of tones that represent markers to 3s to avoid loss of trials instead of ignoring them
-        for i in range(len(highPE)):
-            if highPE[i] == 0:
-                highPE[i] = 3
-            if lowPE[i] == 0:
-                lowPE[i] = 3
-
-        # 1 is standard/bottom, 2 is deviant/high, 3 is "baseline trial"
-
-        stim_types = inputs
-        itis = np.ones_like(inputs) * 0.5
-
-        newAdditionalMarkers = []
-
-        for i in range(0, len(highPE)):
-            newAdditionalMarker = (
-                str(oddball3[i])
-                + str(oddball4[i])
-                + str(oddball5[i])
-                + str(oddball6[i])
-                + str(highPE[i])
-                + str(lowPE[i])
-            )
-            newAdditionalMarkers.append(newAdditionalMarker)
-
-        additional_labels = {"labels": newAdditionalMarkers}
-        aMMN.present(
-            record_duration=record_duration,
-            stim_types=stim_types,
-            itis=itis,
-            additional_labels={"labels": newAdditionalMarkers},
-            eeg=eeg_device,
-            save_fn=save_fn,
-        )
+from eegnb.experiments.auditory_oddball import aob, diaconescu
+from eegnb.experiments.auditory_ssaep import ssaep, ssaep_onefreq
+
+
+experiments = {
+    "visual-N170": n170,
+    "visual-P300": p300,
+    "visual-SSVEP": ssvep,
+    "auditory-SSAEP orig": ssaep,
+    "auditory-SSAEP onefreq": ssaep_onefreq,
+    "auditory-oddball orig": aob,
+    "auditory-oddball diaconescu": diaconescu,
+}
+
+
+def get_exp_desc(exp: str):
+    if exp in experiments:
+        module = experiments[exp]
+        if hasattr(module, "__title__"):
+            return module.__title__  # type: ignore
+    return "{} (no description)".format(exp)
+
+
+def run_experiment(experiment: str, record_duration: float, eeg_device: str, save_fn):
+    if experiment in experiments:
+        module = experiments[experiment]
+        module.present(duration=record_duration, eeg=eeg_device, save_fn=save_fn)  # type: ignore
```

Comment:
This causes issues where the `present` function expects different parameters.

Reply:
This might only be the case for visual_queing though (which wasn't an option before anyway)